request.go

```diff
@@ -327,6 +327,47 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 
 			}
 
+		} else if annotation == annotationLinks {
+			if data.Links == nil {
+				continue
+			}
+
+			links := make(Links, len(*data.Links))
+
+			for k, v := range *data.Links {
+				link := v // default case (including string urls)
+
+				// Unmarshal link objects to Link
+				if t, ok := v.(map[string]interface{}); ok {
+					unmarshaledHref := ""
+					href, ok := t["href"].(string)
+					if ok {
+						unmarshaledHref = href
+					}
+
+					unmarshaledMeta := make(Meta)
+					meta, ok := t["meta"].(map[string]interface{})
+					if ok {
+						for metaK, metaV := range meta {
+							unmarshaledMeta[metaK] = metaV
+						}
+					}
```

Comment:
Can cleanup this as meta is only used inside the condition scope.

```suggestion
					if meta, ok := t["meta"].(map[string]interface{}); ok {
						for metaK, metaV := range meta {
							unmarshaledMeta[metaK] = metaV
						}
					}
```